### PR TITLE
Minimum mod name length #232 

### LINKED ISSUE #232: Minimum map/mod name length
There's a maximum length for map display names https://github.com/FAForever/faf-java-api/blob/develop/src/main/java/com/faforever/api/data/domain/Map.java#L66

there should also be a minimum (we don't want to allow empty string as filename).
I suggest 3.

### PR DESCRIPTION
**Pull request by basalt79**
added also the "nil" check and the missing error tests from the TODO 

would be great to get also the `hacktoberfest-accepted` label

Closes #232

### CODE REVIEW

**Comment by Askaholic** [comment]
Make sure to add ‘Closes #232’ to the pr description.

File: src/test/java/com/faforever/api/mod/ModServiceTest.java [hunk 1]
```diff
@@ -102,58 +105,141 @@ public void processUploadedMod() throws Exception {
...
+      Files.copy(inputStream, uploadFile);
     }
-    return uploadedFile;
+    return uploadFile;
+  }
+
+  @NotNull
+  private Path prepareModDynamic(LuaContent lc) throws IOException {
+    Path uploadFile = temporaryFolder.resolve("uploaded-dynamic-mod.zip");
+    ZipOutputStream zos = new ZipOutputStream(Files.newOutputStream(uploadFile));
```
Comment (bukajsytlos) [inline src/test/java/com/faforever/api/mod/ModServiceTest.java]: please use try-with-resources

**Review by bukajsytlos — CHANGES_REQUESTED** [review]

File: src/test/java/com/faforever/api/mod/ModServiceTest.java (hunk 1 above)
Comment (basalt79) [reply src/test/java/com/faforever/api/mod/ModServiceTest.java]: @bukajsytlos  updated PR

**Review by bukajsytlos — DISMISSED** [review]

File: src/main/java/com/faforever/api/mod/ModService.java [hunk 2]
```diff
@@ -267,7 +277,7 @@ private void store(com.faforever.commons.mod.Mod modInfo, Optional<Path> thumbna
         .setDisplayName(modInfo.getName())
         .setVersions(new ArrayList<>())
         .setUploader(uploader))
-        .setRecommended(false);
+      .setRecommended(false);
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/mod/ModService.java:280]: formatting broke here

File: src/test/java/com/faforever/api/mod/ModServiceTest.java [hunk 3]
```diff
@@ -45,7 +48,7 @@
   @TempDir
   public Path temporaryFolder;
 
-  private ModService instance;
+  private ModService service;
```
Comment (Brutus5000) [inline src/test/java/com/faforever/api/mod/ModServiceTest.java]: the name `instance` is a convention (in this project). please keep it

Comment (basalt79) [reply src/test/java/com/faforever/api/mod/ModServiceTest.java]: reverted name

File: src/main/java/com/faforever/api/mod/ModService.java (hunk 2 above)
Comment (bukajsytlos) [reply src/main/java/com/faforever/api/mod/ModService.java:280]: it is not on the same level

**Review by bukajsytlos — APPROVED** [review]
lgtm

**Pull request merged**